### ISSUE TITLE
Filename Pattern not working as expected.

### ISSUE THREAD

**Issue by martijnvanschie** [rfennell/AzurePipelines#490]
### Azure DevOps Extension you are using
Version Angular Environment.TS File

### Where are you running it?
*You can run the extensions in the cloud or on-premise*
- Azure DevOps Server (on-premise TFS, Version 15.117.26714.0)

### Version of Extension/Task
v2.11.2

### Expected behaviour and actual behaviour
We have multiple environment files used for prod and test (environment.prod.ts). Using the file pattern does not work as expected. This means a patterns like "environment*.ts" will not work and does not find the file.

> 2019-05-22T07:30:21.5881330Z Filename Pattern: environment*.ts
> 2019-05-22T07:30:21.5881330Z Version Number/Build Number: ZBJ Tablet Manager CI 0.4.0.3
> 2019-05-22T07:30:21.5881330Z Version Filter to extract build number: \d+\.\d+\.\d+\.\d+
> 2019-05-22T07:30:21.5881330Z Version Format for JSON File: {1}.{2}.{3}.{4}
> 2019-05-22T07:30:21.5881330Z Field to update (all if empty): version
> 2019-05-22T07:30:21.5881330Z Inject Version: false
> 2019-05-22T07:30:21.5881330Z Output: Version Number Parameter Name: OutputedVersion
> 2019-05-22T07:30:21.5881330Z Extracting version number from build number
> 2019-05-22T07:30:21.5881330Z Extracted Build Version: 0.4.0.3
> 2019-05-22T07:30:21.5881330Z Angular Version Name will be: 0.4.0.3
> 2019-05-22T07:30:26.0256692Z ##[warning]Found no files.

Looking at the code it does an `endsWith` compare and not an actual pattern compare.

```typescript
        if (file.toLowerCase().endsWith(filename.toLowerCase())) {
          filelist.push(path.join(dir, file));
        }
```

### Steps to reproduce the problem

1. Use a node project with multiple environment files. 
2. Use a file pattern of "environment*.ts".
3. Run the build.

**Comment by rfennell** [comment]
Good catch, that shoould be a more complext match, paybe some regrex. I will have a look at it

**Comment by martijnvanschie** [comment]
Looking forward to a fix. These tasks are very handy 👍

**Comment by rfennell** [comment]
@martijnvanschie This has been fixed in 2.12.1

You can now enter the filename as **environmen.ts** regex **environment\.(.*)\.ts**

If this issue remains a problem please do not hesitate to re-open it

**Comment by martijnvanschie** [comment]
Just tested it. Works like a charm.
Many thank :)
👍
